tests/Logging/AbstractListenerTest.php

```diff
@@ -0,0 +1,321 @@
+        $mockArgs = $this->createMock(LifecycleEventArgs::class);
+
+        $mockEntityManager = $this->createMock(EntityManager::class);
+
+        $mockArgs
+            ->expects($this->once())
+            ->method('getEntityManager')
+            ->willReturn($mockEntityManager);
+
+        $mockMunicipality = $this->createMock(Municipality::class);
+
+        $mockArgs
+            ->expects($this->once())
+            ->method('getObject')
+            ->willReturn($mockMunicipality);
+
+        $changeArray = [
+            'name' => [
+                'BeforeName',
+                'AfterName',
+            ],
+        ];
+
+        $mockUnitOfWork = $this->createMock(UnitOfWork::class);
+
+        $mockEntityManager
+            ->expects($this->once())
+            ->method('getUnitOfWork')
+            ->willReturn($mockUnitOfWork);
+
+        $mockUnitOfWork
+            ->expects($this->once())
+            ->method('getEntityChangeSet')
+            ->with($mockMunicipality)
+            ->willReturn($changeArray);
+
+        $mockCaseID = $this->createMock(UuidV4::class);
+
+        $mockCaseIDString = 'testCaseId';
+
+        $mockCaseID
+            ->expects($this->once())
+            ->method('__toString')
+            ->willReturn($mockCaseIDString);
+
+        $mockCase
+            ->expects($this->once())
+            ->method('getId')
+            ->willReturn($mockCaseID);
+
+        $mockMunicipalityID = $this->createMock(UuidV4::class);
+
+        $mockMunicipalityIDString = 'testMunicipalityId';
+
+        $mockMunicipalityID
+            ->expects($this->once())
+            ->method('__toString')
+            ->willReturn($mockMunicipalityIDString);
+
+        $mockMunicipality
+            ->expects($this->once())
+            ->method('getId')
+            ->willReturn($mockMunicipalityID);
+
+        $mockUser = $this->createMock(User::class);
+
+        $this->mockSecurity
+            ->expects($this->once())
+            ->method('getUser')
+            ->willReturn($mockUser);
+
+        $mockUsername = 'TestUser';
+
+        $mockUser
+            ->expects($this->once())
+            ->method('getUsername')
+            ->willReturn($mockUsername);
+
+        $logEntry = $this->mockListener->createLogEntry($testAction, $mockCase, $mockArgs);
+
+        $expectedDataArray = [
+            'name' => 'AfterName',
+        ];
+
+        // Assert all properties are as expected:
+        $this->assertSame($mockCaseIDString, $logEntry->getCaseID());
+        $this->assertSame(get_class($mockMunicipality), $logEntry->getEntityType());
+        $this->assertSame($mockMunicipalityIDString, $logEntry->getEntityID());
+        $this->assertSame($testAction, $logEntry->getAction());
+        $this->assertSame($mockUsername, $logEntry->getUser());
+        $this->assertSame($expectedDataArray, $logEntry->getData());
+    }
+
+    public function testNonSimpleCreateLogEntry()
```

Comment:
This name of the method should maybe be more descriptive?

Reply:
Refactored tests to be more readable and test methods to be more telling